src/main/kotlin/com/terraformation/backend/documentproducer/db/DocumentStore.kt

```diff
@@ -229,14 +229,25 @@ class DocumentStore(
         ?: throw MissingVariableManifestException(documentTemplateId)
   }
 
-  private fun fetchByCondition(condition: Condition? = null): List<ExistingDocumentModel> =
-      dslContext
-          .select(DOCUMENTS.asterisk(), PROJECTS.NAME)
-          .from(DOCUMENTS)
-          .join(PROJECTS)
-          .on(DOCUMENTS.PROJECT_ID.eq(PROJECTS.ID))
-          .where(condition)
-          .fetch()
-          .filter { currentUser().canReadDocument(it[DOCUMENTS.ID]!!) }
-          .map { ExistingDocumentModel.of(it) }
+  private fun fetchByCondition(condition: Condition? = null): List<ExistingDocumentModel> {
+    val lastSavedVersionIdField =
+        with(DOCUMENT_SAVED_VERSIONS) {
+          DSL.field(
+              DSL.select(ID)
+                  .from(DOCUMENT_SAVED_VERSIONS)
+                  .where(DOCUMENT_ID.eq(DOCUMENTS.ID))
+                  .orderBy(CREATED_TIME.desc())
+                  .limit(1))
```

Comment:
How much do we care about this being sorted by created time specifically? An alternate approach that will probably give the same result nearly all the time would be
```kotlin
DSL.select(DSL.max(ID))
    .from(DOCUMENT_SAVED_VERSIONS)
    .where(DOCUMENT_ID.eq(DOCUMENTS.ID))
```
If you wanted to get super fancy, this could also be done with a window function, but that's a bit much for this query, I think.

Reply:
Created time is just a proxy for "the latest one", which is probably served better by the ID as you point out, thanks Steve!